src/components/Map/MapView.tsx

```diff
@@ -107,6 +148,19 @@ export const MapView = () => {
             />
           ))}
 
+        {/* Visualize all waypoints */}
+        {waypoints
+          .filter(
+            (n) =>
+              (!!n.latitudeI && !!n.longitudeI) || // Shows pins with valid latitudes
```

Comment:
What does this condition block? This is fairly confusing to me, and I'm not sure this is doing what you're intending. Also note that it's exceedingly unlikely to get a location of exactly `(0., 0.)` due to GPS noise, even when located at `(0., 0.)`

Reply:
Good catch - I didn't realize latitudeI and longitudeI couldn't be null